runtime/compiler/env/j9methodServer.cpp

```diff
@@ -1355,3 +1355,111 @@ TR_ResolvedJ9JITaaSServerMethod::getCachedResolvedMethod(TR_ResolvedMethodKey ke
       }
    return false;
    }
+
+bool
+TR_ResolvedRelocatableJ9JITaaSServerMethod::storeValidationRecordIfNecessary(TR::Compilation * comp, J9ConstantPool *constantPool, int32_t cpIndex, TR_ExternalRelocationTargetKind reloKind, J9Method *ramMethod, J9Class *definingClass)
+   {
+   TR_J9VMBase *fej9 = (TR_J9VMBase *) comp->fe();
+
+   bool storeClassInfo = true;
+   bool fieldInfoCanBeUsed = false;
+   TR_AOTStats *aotStats = ((TR_JitPrivateConfig *)fej9->_jitConfig->privateConfig)->aotStats;
+   bool isStatic = false;
+
+   TR::CompilationInfo *compInfo = TR::CompilationInfo::get(fej9->_jitConfig);
+   TR_RelocationRuntime *reloRuntime = compInfo->getCompInfoForThread(fej9->vmThread())->reloRuntime();
+
+   isStatic = (reloKind == TR_ValidateStaticField);
+
+   TR_J9ServerVM *feServer = (TR_J9ServerVM *) fej9;
+   J9Class *clazz = (J9Class *) feServer->TR_J9ServerVM::getClassOfMethod((TR_OpaqueMethodBlock *) ramMethod);
```

Comment:
Would it be bad if we called directly feServer->getClassOfMethod()?

Reply:
If I'm not wrong, `fej9` will be instantiated to `TR_SharedCacheServerVM`, so calling `feServer->getClassOfMethod()` will call the AOT of the method, which we do not want.

Reply:
I agree